### ISSUE TITLE
Please check svd validity against CMSIS-SVD.xsd 

### ISSUE THREAD

**Issue by plaes** [NordicSemiconductor/nrfx#129]
While CMSIS standard defines the [format for the SVD files](https://arm-software.github.io/CMSIS_5/SVD/html/svd_Format_pg.html), it also comes with [tools to check the validity of generated SVD](https://arm-software.github.io/CMSIS_5/SVD/html/svd_SVDConv_pg.html).

Validator can be fetched from the official [CMSIS' devtools repository](https://github.com/Open-CMSIS-Pack/devtools/tree/main/tools/svdconv) and the CMSIS-SVD schema validation specification can be found from [ARM's CMSIS_5](https://github.com/ARM-software/CMSIS_5) repository under [CMSIS/Utilities/CMSIS-SVD.xsd](https://github.com/ARM-software/CMSIS_5/blob/develop/CMSIS/Utilities/CMSIS-SVD.xsd).

Having valid SVD files as source can reduce the burden of [inventing hacks](https://github.com/rust-embedded/svd/pull/277) in tooling using SVD's as source of thruth.

Also, it's possible to check validity of SVD files using `xmllint` from libxml2 package on most Linux distributions:
`xmllint --noout --schema CMSIS-SVD.xsd nrf-random-file.svd`

**Comment by Chapoly1305** [comment]
Yeah, I find this issue really annoying. Many tools failed to load those SVD files due to validation issue.
